### PR TITLE
mullvad.openvpn-mullvad: 2.5.3 -> 2.6.0

### LINKED ISSUE #232794: mullvad-2023.3 openvpn plugin loading error
### Describe the bug

After updating to mullvad-2023.3 openvpn connections fail.
Before the update on 2022.5 this was still working fine.

```
OpenVPN 2.5.3 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Jun 17 2021
library versions: OpenSSL 3.0.8 7 Feb 2023, LZO 2.10
PLUGIN_INIT: POST /nix/store/32cim6mwrbzfpyg8wxwzmdar6zab26l5-mullvad-2023.3/share/mullvad/libtalpid_openvpn_plugin.so '[/nix/store/32cim6mwrbzfpyg8wxwzmdar6zab26l5-mullvad-2023.3/share/mullvad/libtalpid_openvpn_plugin.so] [/tmp/talpid-openvpn-d173db26-2152-4035-aeda-fd36f185a7f5]' intercepted=PLUGIN_UP|PLUGIN_ROUTE_UP|PLUGIN_ROUTE_PREDOWN
PLUGIN_INIT: plugin /nix/store/32cim6mwrbzfpyg8wxwzmdar6zab26l5-mullvad-2023.3/share/mullvad/libtalpid_openvpn_plugin.so expressed interest in unsupported plugin types: [want=0x00011005, have=0x0000ffff]
Exiting due to fatal error
```

### Steps To Reproduce
1. update mullvad to 2023.3
2. start mullvad with openvpn enabled

### Expected behavior
OpenVPN connections should work.

### Notify maintainers

@cole-h

### PR DESCRIPTION
**Pull request by TarCV**
###### Description of changes
Update OpenVPN inside Mullvad package to the version (2.6.0) required by the current Mullvad. Fixes #232794 

Also I've copied configure flags from Makefile ([1](https://github.com/mullvad/mullvadvpn-app-binaries/blob/main/Makefile#L17-L20) [2](https://github.com/mullvad/mullvadvpn-app-binaries/blob/main/Makefile#L35)) - as I understand it, Mullvad uses them when compiling their builds.
~~There are also additional flags they add only on Linux, not sure how to add them correctly (see https://github.com/mullvad/mullvadvpn-app-binaries/blob/main/Makefile#L35 ).~~

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
    **Tested against NixOS 23.05 I'm running on my system (510d721ce097150ae3b80f84b04b13b039186571)** 
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/mullvad/openvpn.nix [hunk 1]
```diff
@@ -16,68 +15,82 @@ openvpn.overrideAttrs (oldAttrs:
...
+        commit = "598014de7c063fa4e8ba1fffa01434229faafd04";
+        sha256 = "sha256-+cFX5gmMuG6XFkTs6IV7utiKRF9E47F5Pgo93c+zBXo=";
+      })
+      (fetchMullvadPatch {
+        # "Remove libnsl dep"
+        commit = "845727e01ab3ec9bd58fcedb31b3cf2ebe2d5226";
+        sha256 = "sha256-Via62wKVfMWHTmO7xIXXO7b5k0KYHs1D0JVg3qnXkeM=";
       })
     ];
+    postPatch = oldAttrs.postPatch or [ ] ++ [ "rm ./configure" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mullvad/openvpn.nix:93]:
```suggestion
    postPatch = oldAttrs.postPatch or "" ++ ''
      rm ./configure
    '';
```

**Comment by TarCV** [comment]
Fixed. Also updated configureFlags to match the flags from mullvadvpn-app-binaries on Linux too.
I think this is all that was required for this update.

I see Borg haven't added cole-h as a reviewer and he is the maintainer of this package. So should I ping him?

**Comment by TarCV** [comment]
@cole-h could you please review when you have time?

**Review by cole-h — APPROVED** [review]
If this fixes it for you, then let's get it in. I don't have any way to test right now (I'm not at my computer, nor do I use the OpenVPN tunnels).

File: pkgs/applications/networking/mullvad/openvpn.nix [hunk 2]
```diff
@@ -1,83 +1,108 @@
...
-, autoconf
-, automake
+, libnl
+, autoreconfHook
+, pkg-config
 }:
 
 openvpn.overrideAttrs (oldAttrs:
   let
+    inherit (lib) optional;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mullvad/openvpn.nix:14]: We are using this in two places where we could just have prefixed optional with lib..

**Comment by arashsm79** [comment]
I can also confirm that this has solved the issue with OpenVPN.
Would it be possible to backport this to 23.05?

**Pull request merged**